src/test/conditions/prefix_sha256_test.cpp

```diff
@@ -0,0 +1,225 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/basics/strHex.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/conditions/condition.h>
+#include <ripple/conditions/fulfillment.h>
+#include <ripple/conditions/prefix_sha256.h>
+
+namespace ripple {
+namespace cryptoconditions {
+
+class prefix_sha256_test : public beast::unit_test::suite
+{
+    void check (
+        fulfillment_t const& f,
+        condition_t const& c,
+        Slice test,
+        Slice good)
+    {
+        BEAST_EXPECT (validate (f, c, test) ==
+            ((test == good) && (f.condition() == c)));
+    }
+
+    void testKnown ()
+    {
+        testcase ("Known");
+
+        Slice const empty {};
+        Slice const abc { "abc", 3 };
+        Slice const abcd { "abcd", 4 };
+        Slice const vwxyz { "vwxyz", 5 };
+
+        { // empty prefix with an empty PREIMAGE-SHA256 subfulfillment
```

Comment:
I'd almost want to disallow preimage as a subfulfillment to prefix; it doesn't really make sense. But good to test anyway.